modules/sr/robot/utils.py

```diff
@@ -7,3 +21,30 @@ def map_to_range(
 ) -> float:
     """Maps a value from within one range of inputs to within a range of outputs."""
     return ((value - old_min) / (old_max - old_min)) * (new_max - new_min) + new_min
+
+
+TDevice = TypeVar('TDevice', bound=Device)
```

Comment:
Small thing: I'd encourage that type definitions appear just below imports and above all function definitions and/or constants.